jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -582,6 +671,26 @@ ecma_utf8_string_to_number (const lit_utf8_byte_t *str_p, /**< utf-8 string */
 
       e_in_lit = e_in_lit * 10 + digit_value;
 
+      /**
+       * Sanity check: If the exponent part is bigger than 10^3 we should return with Infinity
+       * because log10(Number.MAX_VALUE) ~= 308.2547.
+       * If the exponent part is less than -10^3  we should return with 0
+       * because log10(Number.MIN_VALUE) ~= -323.3062.
+       * In addition this check prevents e_in_lit overflow even
+       * if the convertible number's exponent part bigger than UINT32_MAX.
+       */
+      if (exponent_length > 3)
```

Comment:
Now I can't agree with you may u got something wrong.
You said 
> E.g., for e0001, exponent_length is 4

But actually is only 1.

```
if (!non_zero_digit_was_seen)
        {
          if (*begin_p != LIT_CHAR_0)
          {
            non_zero_digit_was_seen = true;
          }
          else
          {
            begin_p++;
            continue;
          }
        }
```

This check prevents this issue that you mentioned.

Also the overflow too.
It's true that `e_in_lit` does not store the accurate value of the exponent, but the limit was `10^3`. Your opinion is that it gives false result if the input E.g., `0.00000 .... 1000 times .... 00002`. If the parses would allow that input length, it gave but actually the limit is ` PARSER_MAXIMUM_IDENT_LENGTH 255`. So I can't imagine any input number, which could cause any issue with this part of code.

Reply:
Ah, I've missed the `continue` in the added code. That covers item 1 of my list, indeed. And 308+255 is still 3 digits, true.